vulture_os/authentication/idp/api.py

```diff
@@ -0,0 +1,325 @@
+#!/home/vlt-os/env/bin/python
+"""This file is part of Vulture OS.
+
+Vulture OS is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Vulture OS is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Vulture OS.  If not, see http://www.gnu.org/licenses/.
+"""
+__author__ = "Olivier de Régis"
+__credits__ = []
+__license__ = "GPLv3"
+__version__ = "3.0.0"
+__maintainer__ = "Vulture OS"
+__email__ = "contact@vultureproject.org"
+__doc__ = 'IDP API'
+
+import logging
+from django.views import View
+from django.conf import settings
+from authentication import ldap
+from authentication.ldap import tools
+from django.http import JsonResponse
+from gui.decorators.apicall import api_need_key
+from authentication.ldap import tools
+from django.utils.decorators import method_decorator
+from django.views.decorators.csrf import csrf_exempt
+from django.utils.translation import ugettext_lazy as _
+from authentication.user_portal.models import UserAuthentication
+
+logging.config.dictConfig(settings.LOG_SETTINGS)
+logger = logging.getLogger('api')
+
+def get_repo(portal):
+    ldap_repo = None
+    for repo in portal.repositories.all():
+        if repo.subtype == "LDAP":
+            ldap_repo = repo.get_daughter()
+            break
+
+    if not ldap_repo:
+        # The portal does not have a LDAP repository
+        raise UserAuthentication.DoesNotExist()
+
+    return ldap_repo
+
+
+@method_decorator(csrf_exempt, name="dispatch")
+class IDPApiView(View):
+    @api_need_key("cluster_api_key")
+    def get(self, request, object_id):
+        try:
+            portal = UserAuthentication.objects.get(pk=object_id)
+            ldap_repo = get_repo(portal)
+
+            object_type = request.GET["object_type"].lower()
+            if object_type not in ("groups", "users", "search"):
+                raise KeyError
+
+            if object_type == "groups":
+                data = tools.get_groups(ldap_repo)
+                groups = [elem[ldap_repo.group_attr][0] for elem in data]
+                return JsonResponse({
+                    "data": groups
+                })
+            
+            elif object_type == "users":
+                group_name = request.GET['group_name']
+                group_dn = ldap_repo.create_group_dn(group_name)
+                data = tools.get_users(ldap_repo, group_dn)
```

Comment:
why not let `get_users()` get the group_dn from the group name ?
like it's done with `create_group()`